txlib/api/base.py

```diff
@@ -202,7 +202,12 @@ def __setattr__(self, name, value):
         """
         # If __init__() hasn't finished yet, accept anything
         if ('_is_initialized' not in self.__dict__) or (name in self.__dict__):
-            return super(BaseModel, self).__setattr__(name, value)
+            if name in self.__dict__ and name in self.writable_fields:
+                # Allow fields which are part of the __dict__  function to
+                # be modified
+                self._modified_fields[name] = value
+            else:
+                return super(BaseModel, self).__setattr__(name, value)
```

Comment:
Why not simply bring the second check above? Wouldn't this work as expected, and without any side effects?

```python
if name in self.writable_fields:
    self._modified_fields[name] = value
elif ('_is_initialized' not in self.__dict__) or (name in self.__dict__):
    return super(BaseModel, self).__setattr__(name, value)
else:
    self._handle_wrong_field(name, ATTR_TYPE_WRITE)
```

Reply:
I don't think that this will work.

The class has an initialisation logic (the `_is_initialized ` flag) that needs to be triggered first. If we do it the way you mention, `name in self.writable_fields` will always be true and called before the initialisation part, causing who knows what issues.

As an example, `slug` is part of the `writable_fields ` and also part of the `self.__dict__`. With the above code, `return super(BaseModel, self).__setattr__(name, value)` will never run if `name = 'slug'`.

Reply:
You might be right. The code is not very straightforward and I don't want us to introduce any bugs, so I'm OK with your implementation. Ideally, we should make this simpler in the future, as it's hard to follow and maintain.